### PR TITLE
email embed parent component route

### PR DESCRIPTION
**Pull request by dblatcher**
## What does this change?
The Marketing Tools team is adding newsletter signup in the Epics and intends to use the same iframes that are used in the newsletter sign-up embeds. So the tracking events can distinguish between interactions with an iframe in an epic and interactions with an iframe in an embed, this PR:
 - adds extra routes for rendering the email embed pages with an optional parentComponent path param
 - passes the value of that param to the email sign-up form template to be used within the `data-component` attributes which set the tracking event payload posted to ophan

## Does this change need to be reproduced in dotcom-rendering ?

- [x] No
- [ ] Yes (please indicate your plans for DCR Implementation)

## What is the value of this and can you measure success?
When Marketing Tools implement sign-ups in epics, they will be able to use the new route for the iframe src URL, so clicks from the epic can be separated from clicks from embeds.

### Does this affect other platforms?

- [ ] AMP 
- [ ] Apps
- [ ] Other (please specify)

### Does this affect GLabs Paid Content Pages? Should it have support for Paid Content?

- [x] No
- [ ] Yes (please give details)

### Does this change break ad-free?

- [x] No
- [ ] It did, but tests caught it and I fixed it
- [ ] It did, but there was no test coverage so I added that then fixed it

### Does this change update the version of CAPI we're using?

- [x] No, all the existing database files are just fine
- [ ] Yes, and I have [re-run all the tests locally and checked in all the updated data/database/xyz files](https://github.com/guardian/frontend/blob/main/docs/03-dev-howtos/15-updating-test-database.md)

### Accessibility test checklist

- [ ] [Tested with screen reader](https://accessibility.gutools.co.uk/testing/web/screen-readers/)
- [ ] [Navigable with keyboard](https://accessibility.gutools.co.uk/testing/web/keyboard-navigation/)
- [ ] [Colour contrast passed](https://accessibility.gutools.co.uk/testing/web/colour-contrast/)

### Tested

- [x] Locally
- [ ] On CODE (optional)

### CODE REVIEW

File: common/app/controllers/EmailSignupController.scala [hunk 1]
```diff
@@ -190,7 +190,11 @@ class EmailSignupController(
       }
     }
 
-  def renderForm(emailType: String, listId: Int): Action[AnyContent] =
+  def renderFormWithParentComponent (emailType: String, listId: Int, parentComponent: String): Action[AnyContent] =
+    return renderForm(emailType,listId, Option(parentComponent))
```
Comment (AshCorr) [inline common/app/controllers/EmailSignupController.scala:194]:
```suggestion
    renderForm(emailType,listId, Option(parentComponent))
```

I think the return is unecessary here maybe?

**Review by AshCorr — APPROVED** [review]
Looks good to me! Just one or two nitpicky comments

File: common/app/views/fragments/email/signup/subscription/emailSignUp.scala.html [hunk 2]
```diff
@@ -35,6 +36,9 @@
 @formClass = @{ "email-sub__form" + " email-sub__form--" + componentClass }
 @headerClass = @{"email-sub__header" + " email-sub__header--" + componentClass  }
 
+@parentSuffix = @{ if (iframeParentComponent.nonEmpty) " " + iframeParentComponent.getOrElse("") else "" }
```
Comment (AshCorr) [inline common/app/views/fragments/email/signup/subscription/emailSignUp.scala.html:39]:
```suggestion
@parentSuffix = @{iframeParentComponent.map(" " + _).getOrElse("")}
```

Maybe?

**Comment by dblatcher** [comment]
> Looks good to me! Just one or two nitpicky comments

Great - thanks for the improvements!

**Pull request merged**